bld/CLMBuildNamelist.pm

```diff
@@ -1982,10 +1983,82 @@ sub setup_logic_decomp_performance {
 
 #-------------------------------------------------------------------------------
 
+sub setup_logic_snicar_methods {
+  my ($opts, $nl_flags, $definition, $defaults, $nl) = @_;
+
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_snw_shape' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_solarspec' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_dust_optics' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_numrad_snw' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_snobc_intmix' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_snodst_intmix' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_use_aerosol' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'snicar_aerforc_diag' );
+  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'do_sno_oc' );
+  # Error checking
+  my $opt1 = 'snicar_snw_shape';
+  my $var1 = $nl->get_value($opt1);
+  my $sup1a = "'sphere'";  # supported value for this option
+  my $sup1b = "'hexagonal_plate'";  # supported value for this option
+  if (($var1 ne $sup1a) && ($var1 ne $sup1b)) {
+    $log->warning("$opt1=$sup1a and $sup1b are supported; $var1 is EXPERIMENTAL, UNSUPPORTED, and UNTESTED!");
+  }
+  my $opt2 = 'snicar_solarspec';
```

Comment:
@slevis-lmwg and I talked about this and we'd like to change this into a loop for maintainability.

Reply:
@slevis-lmwg there's an example of a warning inside a loop in check_megan_spec the subroutine.

An example of a dictionary in perl is the %opts hash. You then use the keys function inside a foreach loop

so something like...
``` perl
     my %supportedSettings = ( 'thing' => '.false.' );
      foreach my $key (keys %supportedSettings) {
          if ( $nl->get_value($key) ne $supportedSettings{$key} ) {
               $log->warning("....");
          }
      }
```